### PR TITLE
TensorFlow into ClickHouse

### PR DESCRIPTION
**Pull request by Provet**
Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Allow TensorFlow Lite models evaluation

Detailed description / Documentation draft:
Integrate [TensorFlow Lite C API](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/c/README.md) into modelEvaluate function via shared library.

### CODE REVIEW

**Comment by kitaisreal** [comment]
@Provet need to add integration test.

File: src/Interpreters/ExternalModelsLoader.cpp [hunk 1]
```diff
@@ -34,6 +35,18 @@ std::shared_ptr<const IExternalLoadable> ExternalModelsLoader::create(
                 lifetime
         );
     }
+    else if (type == "tensorflow")
+    {
+        int32_t num_threads = 1;
```
Comment (kitaisreal) [inline src/Interpreters/ExternalModelsLoader.cpp:40]: It is better to do interpreteration in 1 thread, we does not prefer if external libraries create threads that we cant control.

File: src/Interpreters/TensorFlowModel.cpp [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+                if (!column->isNumeric())
+                    throw Exception("Unsupported non-numeric data types", ErrorCodes::BAD_ARGUMENTS);
+
+                auto copyToTensor = [this, &tensor](const auto& value) {
+                    if (api->TfLiteTensorCopyFromBuffer(tensor, &value, sizeof(value)) != TfLiteStatus::kTfLiteOk) 
+                        throw Exception("Copy to tensor was failed", ErrorCodes::BAD_ARGUMENTS);
+                };
+
+                auto type = api->TfLiteTensorType(tensor);
+                switch(type)
```
Comment (kitaisreal) [inline src/Interpreters/TensorFlowModel.cpp:267]: It seems that we copy into tensor by 1 element ? Better to get raw data from column and put it in one call.

**Comment by qoega** [comment]
@Mergifyio update

File: src/Interpreters/TensorFlowModel.cpp [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+        
+        if (columns.size() != static_cast<size_t>(tensors_count))
+        {
+            std::string msg;
+            {
+                WriteBufferFromString buffer(msg);
+                buffer << "Number of columns is different with number of tensors: ";
+                buffer << columns.size() << " vs " << tensors_count;
+            }
+            throw Exception(msg, ErrorCodes::BAD_ARGUMENTS);
```
Comment (vdimir) [inline src/Interpreters/TensorFlowModel.cpp:218]:
```suggestion
            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Number of columns is different with number of tensors: {} and {}", columns.size(), tensors_count);
```
```

**Comment by alexey-milovidov** [comment]
Too less number of likes for this pull request.

**Pull request closed without merging**